src/main/java/seedu/duke/commands/AddCommand.java

```diff
@@ -132,37 +144,59 @@ public String[] splitDetails(String details) {
         String quantity;
         String unit;
 
-        String[] temp = details.split(QUANTITY_SEPARATOR);
+        String[] temp = details.trim().split(QUANTITY_SEPARATOR, 2);
+
         String[] unitTemp = details.split(UNIT_SEPARATOR);
 
         if (temp[0].indexOf("-n") < temp[0].indexOf("-e")) {
-            String[] nameAndDate = temp[0].replace(NAME_SEPARATOR, "").split(EXPIRY_SEPARATOR, 2);
-            name = nameAndDate[0];
-            date = nameAndDate[1];
+            String[] nameAndDate = temp[0].replace(NAME_SEPARATOR, "").trim().split(EXPIRY_SEPARATOR, 2);
+            name = nameAndDate[0].trim();
+            date = nameAndDate[1].trim();
         } else {
-            String[] dateAndName = temp[0].replace(EXPIRY_SEPARATOR, "").split(NAME_SEPARATOR, 2);
-            name = dateAndName[1];
-            date = dateAndName[0];
+            String[] dateAndName = temp[0].replace(EXPIRY_SEPARATOR, "").trim().split(NAME_SEPARATOR, 2);
+            name = dateAndName[1].trim();
+            date = dateAndName[0].trim();
         }
 
         if (hasQuantity && !hasUnit) {
-            quantity = temp[1];
+            quantity = temp[1].trim();
             String[] foodDetails = {name, date, quantity};
             return foodDetails;
         }
 
-        if(hasUnit){
-            String[] quantityTemp = unitTemp[0].replace(UNIT_SEPARATOR, "").split(QUANTITY_SEPARATOR);
+        if (hasUnit) {
+            String[] quantityTemp = unitTemp[0].replace(UNIT_SEPARATOR, "").split(QUANTITY_SEPARATOR, 2);
             quantity = quantityTemp[1].trim();
-            String unit_temp = unitTemp[1].trim();
-            double quantityInDouble = Double.parseDouble(quantity);
-            unit = getUnitOfFood(unit_temp, quantityInDouble);
-
+            String unitTemporary = unitTemp[1].trim();
+            Double quantityInDouble = Double.parseDouble(quantity);
+            unit = getUnitOfFood(unitTemporary, quantityInDouble);
             String[] foodDetails = {name, date, quantity, unit};
             return foodDetails;
         }
 
         String[] foodDetails = {name, date};
         return foodDetails;
     }
+
+    /**
+     * Returns whether the input date is a valid date
+     *
+     * @param date the date String
+     * @return isValid whether the date is valid
+     */
+    public boolean isValidDate(String date) {
+        LocalDate currentDate = LocalDate.now();
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+
+        try {
+            LocalDate expiryDate = LocalDate.parse(date, formatter);
+            boolean isValid = expiryDate.isAfter(currentDate);
+            return isValid;
+        } catch (DateTimeParseException e) {
+            System.out.println("The input date is not valid");
```

Comment:
Instead of not valid, can say trying to add expired product.

Reply:
I have updated the error message above!! Thanks!!